src/components/sidebar/index.tsx

```diff
@@ -1,52 +1,94 @@
 import { useEffect } from 'react'
 import { useDispatch, useSelector } from 'react-redux'
 
-import { getCurrentLocationForecast } from '../../store/action'
 import { RootState } from '../../store/root-reducer'
+import { getForecast } from '../../store/action'
+import { getCurrentLocationForecast } from '../../api'
 
-import s from './index.module.css'
+import ShowerIcon from '../../assets/images/shower.png'
+import CloudIcon from '../../assets/images/cloudy.png'
 
-const Sidebar = () => {
+import {
+  SidebarContaner,
+  Header,
+  SearchButton,
+  GeolocationButton,
+  Body,
+  SmallCloud,
+  MediumCloudRight,
+  MediumCloudLeft,
+  LargeCloud,
+  WeatherIcon,
+  Temperature,
+  TemperatureUnit,
+  WeatherStateName,
+  Footer,
+  CurrentDate,
+  Location,
+  VectorIcon,
+  CurrentLocation
+} from './style'
+
+const Sidebar: React.FC = (): JSX.Element => {
   const dispatch = useDispatch()
-  const { forecastData } = useSelector((state: RootState) => state.forecastData)
 
+  useEffect((): any => {
+    currentLocationForecast()
+  }, [])
+
+  const { forecastData } = useSelector((state: RootState) => state.forecastData)
   const { title, consolidated_weather } = forecastData
   const { weather_state_name, applicable_date, the_temp } =
     consolidated_weather[0]
 
-  useEffect(() => {
-    console.log('testFirstRender')
-    currentLocationForecast()
-  }, [])
+  const fixedTemp: number = parseFloat(the_temp.toFixed(1))
+
+  const currentLocationForecast = (): any => {
+    getCurrentLocationForecast().then((data: any) => {
+      dispatch(getForecast(data))
+    })
+  }
 
-  const currentLocationForecast = () => dispatch(getCurrentLocationForecast())
   return (
-    <div className={s.sidebar}>
-      <div className={s.header}>
-        <button className={s.searchButton}>Search for place</button>
-        <button className={s.geolocationButton}>O</button>
-      </div>
-      <div className={s.body}>
-        <img
-          className={s.weatherIcon}
-          src="https://s3-alpha-sig.figma.com/img/f823/173c/6d67db3d534c1d812e09aed0501cdb6e?Expires=1632700800&Signature=N-5-EDzGVNnweLD9F1swh0n2ZBNLF7ZvDMCyX3BTCNp~uZ4MD4l7v8RallOA0tiWjoPdAmBvJRdMBcnylybAFCEkDnp9IlHPX38DHYD0VuoHmjwh98q4~FMPhfxHw1Z5GNUZXtMIq6fg3NalgOWNVSL8wKcLlGxtjJS637rWiKIvXbYgP-FotVg4aT~3JW0Z5WvouPwDV4WBX2vG1Jw1hLyRkNMrokcNGiVxVg6kM12dTaG8YwEvvKDd0xs5Vofpdp2fSNR~KoZ8-brdDEv26fKCrc~Ro8UEFhRLPLSAfpYEbjgXTVqtfyTMV7b7oPEOaKeWR93q40Ye64n7PNJOHQ__&Key-Pair-Id=APKAINTVSUGEWH5XD5UA"
-          alt="weather-icon"
+    <SidebarContaner>
+      <Header>
+        <SearchButton>Search for place</SearchButton>
+        <GeolocationButton>O</GeolocationButton>
```

Comment:
can be done as separate badge component in folder 'common'
find and use current location icon instead of `0`